File: app/src/schemas/satellite.py
"""Satellite analysis schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SatelliteAnalysisResponse(BaseModel):
    id: str
    property_id: str
    water_feature_id: Optional[str] = None
    pool_detected: bool
    estimated_pool_sqft: Optional[float] = None
    pool_confidence: float
    vegetation_pct: float
    canopy_overhang_pct: float
    hardscape_pct: float
    shadow_pct: float
    pool_lat: Optional[float] = None
    pool_lng: Optional[float] = None
    image_url: Optional[str] = None
    image_zoom: int
    analysis_version: str
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SetPinRequest(BaseModel):
    pool_lat: float
    pool_lng: float


class AnalyzeRequest(BaseModel):
    pool_lat: Optional[float] = None
    pool_lng: Optional[float] = None
    force: bool = False


class PoolBowWithCoordsResponse(BaseModel):
    id: str
    property_id: str
    bow_name: Optional[str] = None
    water_type: str
    address: str
    city: str = ""
    customer_id: str
    customer_name: str
    customer_type: str = "residential"
    pool_sqft: Optional[float] = None
    lat: Optional[float] = None
    lng: Optional[float] = None
    pool_lat: Optional[float] = None
    pool_lng: Optional[float] = None
    has_analysis: bool = False
    tech_name: Optional[str] = None
    tech_color: Optional[str] = None


class BulkAnalysisRequest(BaseModel):
    wf_ids: Optional[list[str]] = None
    property_ids: Optional[list[str]] = None  # deprecated fallback
    force_reanalyze: bool = False


class BulkAnalysisResponse(BaseModel):
    total: int
    analyzed: int
    skipped: int
    failed: int
    results: list[SatelliteAnalysisResponse]

File: frontend/src/types/satellite.ts
export interface SatelliteAnalysis {
  id: string;
  property_id: string;
  water_feature_id: string | null;
  pool_detected: boolean;
  estimated_pool_sqft: number | null;
  pool_confidence: number;
  vegetation_pct: number;
  canopy_overhang_pct: number;
  hardscape_pct: number;
  shadow_pct: number;
  pool_lat: number | null;
  pool_lng: number | null;
  image_url: string | null;
  image_zoom: number;
  analysis_version: string;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface PoolBowWithCoords {
  id: string;
  property_id: string;
  wf_name: string | null;
  water_type: string;
  address: string;
  city: string;
  customer_id: string;
  customer_name: string;
  customer_type: string;
  pool_sqft: number | null;
  lat: number | null;
  lng: number | null;
  pool_lat: number | null;
  pool_lng: number | null;
  has_analysis: boolean;
  tech_name: string | null;
  tech_color: string | null;
}

export interface BulkAnalysisRequest {
  wf_ids?: string[] | null;
  force_reanalyze?: boolean;
}

export interface BulkAnalysisResponse {
  total: number;
  analyzed: number;
  skipped: number;
  failed: number;
  results: SatelliteAnalysis[];
}
